File: API/Entities/LoanApplication.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace API.Entities;

public class LoanApplication
    {
        public int Id { get; set; }

        [Required]
        public int LoanOfferId { get; set; }
        public LoanOffer LoanOffer { get; set; }

        [Required]
        public int BorrowerId { get; set; }
        public User Borrower { get; set; }
        public decimal MonthlyRepayment {get;set;}
        public decimal TotalRepayment {get;set;}

        [Required]
        public LoanApplicationStatus Status { get; set; }

        public DateTime AcceptanceDate { get; set;}

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    }

      public enum LoanApplicationStatus
    {
        Pending,
        Approved,
        Rejected,
        Withdrawn
    }

File: client/src/app/_models/loanApplication.ts
// src/app/models/loan-application.model.ts

import { LoanOffer } from "./loan-offer";
import { Payment } from "./payment";

export enum LoanApplicationStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
  Withdrawn = 3
}
  
export interface LoanApplication {
    id: number;
    loanOfferId: number;
    loanOffer: LoanOffer;
    borrowerId: number;
    monthlyRepayment:number;
    totalRepayment:number;
    borrower: Borrower;
    status: LoanApplicationStatus;
    acceptanceDate: string;
    createdAt: string;
    updatedAt: string;
    payments: Payment[];
  }
  export interface Borrower {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    passwordHash: string;
    walletAddress: string;
    creditScore: number;
    idNumber: string;
    address: string;
    phoneNumber: string;
    employmentStatus: string;
    annualIncome: number;
    role: number;
    createdAt: string;
    updatedAt: string;
    loanOffers?: any;
    ratings: any[];
  }
